src/modules/SimpleSlippageRouter.sol

```diff
@@ -0,0 +1,145 @@
+     * @param maxShares specified acceptable max shares amount.
+     * @param minimumAssets specified minimum amount of assets to be returned.
+     * @param actualAssetsQuoted actual amount of assets to come from proposed tx.
+     */
+    error SimpleSlippageAdaptor__RedeemMinAssetsUnmet(
+        uint256 maxShares,
+        uint256 minimumAssets,
+        uint256 actualAssetsQuoted
+    );
+
+    /**
+     * @notice deposits assets into specified cellar w/ _minimumShares expected and _deadline specified.
+     * @param _cellar specified cellar to deposit assets into.
+     * @param _assets amount of cellar base assets to deposit.
+     * @param _minimumShares amount of shares required at min from tx.
+     * @param _deadline block.timestamp that tx must be carried out by.
+     */
+    function deposit(Cellar _cellar, uint256 _assets, uint256 _minimumShares, uint64 _deadline) public {
+        if (block.timestamp > _deadline) revert SimpleSlippageAdaptor__ExpiredDeadline(_deadline);
+        ERC20 baseAsset = _cellar.asset();
+        baseAsset.safeTransferFrom(msg.sender, address(this), _assets);
+        uint256 shares = _cellar.previewDeposit(_assets);
+        if (shares < _minimumShares) revert SimpleSlippageAdaptor__DepositMinimumSharesUnmet(_minimumShares, shares);
+        baseAsset.approve(address(_cellar), _assets);
+        _cellar.deposit(_assets, msg.sender);
+        _revokeExternalApproval(baseAsset, address(_cellar));
+    }
+
+    /**
+     * @notice withdraws assets as long as tx returns more than _assets and is done before _deadline.
+     * @param _cellar specified cellar to withdraw assets from.
+     * @param _assets amount of cellar base assets to withdraw.
+     * @param _maxShares max amount of shares to redeem from tx.
+     * @param _deadline block.timestamp that tx must be carried out by.
+     */
+    function withdraw(Cellar _cellar, uint256 _assets, uint256 _maxShares, uint64 _deadline) public {
+        if (block.timestamp > _deadline) revert SimpleSlippageAdaptor__ExpiredDeadline(_deadline);
+
+        uint256 shares = _cellar.previewWithdraw(_assets);
+        if (shares > _maxShares) revert SimpleSlippageAdaptor__WithdrawMaxSharesSurpassed(_maxShares, shares);
+
+        _cellar.withdraw(_assets, msg.sender, msg.sender); // NOTE: user needs to approve this contract to spend shares
+    }
+
+    /**
+     * @notice mints shares from the cellar and returns shares to receiver IF shares quoted cost are less than specified _assets amount by the specified _deadline.
+     * @param _cellar specified cellar to deposit assets into.
+     * @param _maxAssets max amount of cellar base assets to deposit.
+     * @param _shares amount of shares required at min from tx.
+     * @param _deadline block.timestamp that tx must be carried out by.
+     */
+    function mint(Cellar _cellar, uint256 _maxAssets, uint256 _shares, uint64 _deadline) public {
```

Comment:
For this function, and redeem below, I think it makes more sense to swap the assets argument and the shares argument.

That way for every function the arguments will be
cellar, value needed for cellar call, slippage value, deadline